patterns/nvidia-gpu-efa/README.md

```diff
@@ -82,84 +88,178 @@ See [here](https://aws-ia.github.io/terraform-aws-eks-blueprints/getting-started
     clusterrole.rbac.authorization.k8s.io/mpi-operator configured
     ```
 
-3. EFA test
+3. EFA info test
 
-    The results should shown that two EFA adapters are available (one for each worker pod)
+    This test prints a list of available EFA interfaces by using the `/opt/amazon/efa/bin/fi_info` utility.
+    Edit the script [generate-efa-info-test.sh](generate-efa-info-test.sh) and adjust the following environment variables if needed prior to running it:
+    
+    ```sh
+    export NUM_WORKERS=2
+    export GPU_PER_WORKER=8
+    export EFA_PER_WORKER=32
+    ```
+    
+    `NUM_WORKERS` - number of nodes you want to run the test on
+    `GPU_PER_WORKER` - number of GPUs available on each node
+    `EFA_PER_WORKER` - number of EFA interfaces available on each node
+    
+    ```sh
+    ./generate-efa-info-test.sh
+    ```
+    
+    This script generates an MPIJob manifest file named `efa-info-test.yaml`.
```

Comment:
we should just provide the static yaml manifest, not a script to dynamically create one

Reply:
The manifests are verbose and would make the readme harder to read, also copying and pasting yaml could introduce alignment errors. We could have the static yaml instead of script, but the difference is minimal. In my opinion, the script helps point out the different settings of the yaml where changes can occur and what their meaning is. The script works without change for two P5 instances, but can easily be adapted to other scenarios if needed, which is also valuable. Would like to keep this set up if acceptable.